lntest/itest/lnd_test.go

```diff
@@ -9250,13 +9470,34 @@ out:
 			"instead failed due to: %v", resp.PaymentError)
 	}
 
-	// We'll also ensure that the encoded error includes the invlaid HTLC
+	// We'll also ensure that the encoded error includes the invalid HTLC
 	// amount.
 	if !strings.Contains(resp.PaymentError, htlcAmt.String()) {
 		t.Fatalf("error didn't include expected payment amt of %v: "+
 			"%v", htlcAmt, resp.PaymentError)
 	}
 
+	// Alice sent the failed payment, se we expect her to have a forward event
+	// and a forwarding failure.
+	expectedHash = [][]byte{carolInvoice.RHash}
+	if err := testHTLCEvents(aliceCl, expectedHash, nil, nil, 1); err != nil {
+		t.Fatalf("alice did not have expected HTLC events: %v", err)
+	}
+
+	// Bob forwarded the failed payment, so we expect him to have a forward
+	// event and a forwarding failure.
+	if err := testHTLCEvents(bobCl, expectedHash, nil, nil, 1); err != nil {
+		t.Fatalf("bob did not have expected HTLC events: %v", err)
+	}
+
+	// Carol was the recipient of a payment with an incorrect amount, so
+	// we expect her to have a link failure event with reason incorrect amount.
+	// TODO(carla): replace with specific error once available.
```

Comment:
This TODO is here intentionally, as is errors for invoice payment attempts which fail due to underpayment or expiry too soon are silenced and the `processExitHop` code falls through to `processHodlEvent` which treats these failures as a hodl cancel. 

This structure is being changed in #3770, which is currently in review. Once that change is in, reasons for invoice failure can be better surfaced and this test can be updated.